types/ckeditor__ckeditor5-engine/src/conversion/conversion.d.ts

```diff
@@ -28,5 +28,5 @@ export default class Conversion {
     elementToElement(definition: ConverterDefinition): void;
     for(groupName: 'dataDowncast' | 'editingDowncast' | 'downcast'): DowncastHelpers;
     for(groupName: 'upcast'): UpcastHelpers;
-    for(groupName: string): DowncastHelpers | UpcastHelpers;
+    for(groupName: 'upcast' | 'downcast' | 'dataDowncast' | 'editingDowncast'): UpcastHelpers | DowncastHelpers;
```

Comment:
Isn't this just duplicating what is above? The original `groupName: string` was there for those cases when the function is called with an unknown string.

Reply:
The only issue with `groupName: string` is that it's incorrect. If you called it with an arbitrary string it will throw a [conversion-for-unknown-group](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-conversion-for-unknown-group) CKEditorError. That said, I'm fine with having that be a runtime error and not a type error if you think that's better.

Reply:
Does CKEditor allow creating new `groupName`s? If so, then we should keep the `string`. Otherwise, maybe just nothing would be fine.